testtools/mock_logging.go

```diff
@@ -0,0 +1,43 @@
+package testtools
+
+type callingFatalOnErrorFuncStats struct {
+	FatalOnErrorCallsCount int
+	Err error
+}
+
+type callingPrintFuncStats struct {
+	PrintLnCallsCount int
+	PrintMsg string
+}
+
+type infoLoggerMock struct {
+	Stats *callingPrintFuncStats
+}
+
+func (loggerMock infoLoggerMock) Println(v ...interface{}) {
+	loggerMock.Stats.PrintLnCallsCount++
+	loggerMock.Stats.PrintMsg = v[0].(string)
+}
+
+type errorLoggerMock struct {
+	Stats *callingFatalOnErrorFuncStats
+}
+
+func (loggerMock errorLoggerMock) FatalOnError(err error) {
+	loggerMock.Stats.FatalOnErrorCallsCount++
+	loggerMock.Stats.Err = err
+}
+
+func MockLoggers() (infoLoggerMock, errorLoggerMock) {
+	infoStats := callingPrintFuncStats{
+		PrintLnCallsCount: 0,
+		PrintMsg: "",
+	}
+	errorStats := callingFatalOnErrorFuncStats{
+		FatalOnErrorCallsCount: 0,
+		Err: nil,
+	}
+
+	return infoLoggerMock{ Stats: &infoStats },
```

Comment:
May be it's better to return Logging struct (infoLogger, errorLogger)?

Reply:
Well, inside a test we can't get infoLogger and errorLogger as mock structs (with usefull Stats fields for tests) because they are interfaces and have only Println and FatalOnError methods, so I think it wouldn't be better...